--- lib/geodyn/UTCTime.cpp.orig	2013-06-26 11:08:32.000000000 -0500
+++ lib/geodyn/UTCTime.cpp	2013-08-08 14:20:29.000000000 -0500
@@ -364,25 +364,12 @@
       // GPS time to UTC time
       // @param gpst    GPST as input 
       // @param utc     UTC as output
-   void GPST2UTC(const UTCTime& gpst, UTCTime& utc)
+   void GPST2UTC(const CommonTime& gpst, UTCTime& utc)
    {
-      UTCTime T;
-
-      double mjdGPST = static_cast<Epoch>(gpst).MJD();
-      double mjdTAI = mjdGPST+ UTCTime::TAI_GPS/UTCTime::DAY_TO_SECOND;
-
-         // input should be utc
-      double dtu = IERS::TAImUTC(mjdTAI);               
-      double mjdUTC = mjdTAI - dtu / UTCTime::DAY_TO_SECOND;
-
-      dtu = IERS::TAImUTC(mjdUTC);
-      mjdUTC = mjdTAI - dtu / UTCTime::DAY_TO_SECOND;      
-
-      utc = gpst;                           // GPST
-      utc += UTCTime::TAI_GPS;            // TAI
-      utc += -IERS::TAImUTC(mjdUTC);      // UTC
-
+      CommonTime gps_ct = gpst; 
+      utc = UTCTime(gps_ct);             // utc
    }
+
       // UTC time to GPS time 
       // @param utc    UTC as input 
       // @param gpst   GPST as output 
@@ -422,7 +409,7 @@
       // TT time to UTC time
       // @param tt      TT as input
       // @param utc     UTC as output 
-   void TT2UTC(const CommonTime& tt, CommonTime& utc)
+   void TT2UTC(const CommonTime& tt, UTCTime& utc)
    {
       UTCTime T(static_cast<YDSTime>(tt).year,static_cast<YDSTime>(tt).doy,static_cast<YDSTime>(tt).sod);
       
